### DISCUSSION TITLE
Multipart upload with pre-signed URL?

### DISCUSSION THREAD

**Discussion by sdavis-cat** [soto-project/soto#484] [Q&A]
I've been trying to get multipart upload working with a pre-signed URL but it doesn't seem possible. Though it's possible that I'm missing something, I am new to Soto and multipart uploading.

In my app(MacOS) I will be passed a pre-signed URL to be used for multipart upload but in order to call `s3.multipartUpload()` a `CreateMultipartUploadRequest` must be created.

To create a `CreateMultipartUploadRequest` a bucket is necessary but I don't have a bucket or credentials(just the URL).

I have tried passing in the pre-signed URL as the bucket to `CreateMultipartUploadRequest` when I call `multipartUpload` I get the error `SotoCore.CredentialProviderError error 1`

I am using Soto 5.3.0, MacOS 10.15.7, a pre-signed URL for authentication(which is not created via soto)

The sample code I have seen assumes that the pre-signed URL is created with Soto but this is not the case in my situation(it is return from a web request). If someone could point me to a sample of this it would be much appreciated.

thanks!

**Comment by adam-fowler** [comment]
If you are being provided with signed URLs you shouldn't need Soto. You just need a HTTP client to POST the URL.
